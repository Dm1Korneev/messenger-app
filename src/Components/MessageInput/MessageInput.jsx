import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

const MessageInput = ({ onSendMessage }) => {
  const [messageText, setMessageText] = useState('');

  const handleChange = (event) => {
    setMessageText(event.target.value);
  };

  const handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      event.preventDefault();
      if (messageText.trim()) {
        onSendMessage({ messageText });
        setMessageText('');
      }
    }
  };

  return (
    <TextField
      value={messageText}
      onChange={handleChange}
      onKeyPress={handleKeyPress}
      multiline
      rows="2"
      rowsMax="5"
      fullWidth
      margin="normal"
      label="Write a message"
      variant="outlined"
    />
  );
};

MessageInput.propTypes = {
  onSendMessage: PropTypes.func.isRequired,
};

export default MessageInput;

