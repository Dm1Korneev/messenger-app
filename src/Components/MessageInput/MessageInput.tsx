import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import { useDispatch } from 'react-redux';

import { sendMessage } from 'Redux/actions';

export const MessageInput = () => {
  const dispatch = useDispatch();

  const [messageText, setMessageText] = useState('');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setMessageText(event.target.value);
  };

  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      event.preventDefault();
      if (messageText.trim()) {
        dispatch(sendMessage({ messageText }));
        setMessageText('');
      }
    }
  };

  return (
    <TextField
      value={messageText}
      onChange={handleChange}
      onKeyPress={handleKeyPress}
      multiline
      rows="2"
      rowsMax="5"
      fullWidth
      margin="normal"
      label="Write a message"
      variant="outlined"
    />
  );
};
