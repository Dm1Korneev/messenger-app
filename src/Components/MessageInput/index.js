import commonHoc from 'Components/commonHoc';

import { sendMessage } from 'Redux/actions';
import MessageInput from './MessageInput';

const mapDispatchToProps = {
  onSendMessage: sendMessage,
};

export default commonHoc(MessageInput, {
  mapDispatchToProps,
});

