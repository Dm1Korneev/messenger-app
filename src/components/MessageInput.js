import React, { Component } from "react";
import "./MessageInput.css";
import TextField from "@material-ui/core/TextField";

class MessageInput extends Component {
  constructor(props) {
    super(props);
    this.state = { messageText: "" };
  }

  handleChange = event => {
    this.setState({ messageText: event.target.value });
  };

  handleKeyPress = event => {
    if (event.key === "Enter" && this.state.messageText) {
      event.preventDefault();
      this.props.onSendMessage(this.state.messageText);
      this.setState({ messageText: "" });
    }
  };

  render() {
    const { messageText } = this.state;

    return (
      <div className="MessageInput">
        <TextField
          value={messageText}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
          color="primary"
          multiline
          rows="2"
          rowsMax="5"
          fullWidth
          margin="normal"
          label="Write a message..."
          variant="outlined"
        />
      </div>
    );
  }
}

export default MessageInput;
