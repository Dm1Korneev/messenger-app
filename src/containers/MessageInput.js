import { connect } from 'react-redux';
import { sendMessage } from 'Redux/actions';
import MessageInput from 'Components/MessageInput';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  onSendMessage: (messageText) => dispatch(sendMessage(messageText)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MessageInput);
