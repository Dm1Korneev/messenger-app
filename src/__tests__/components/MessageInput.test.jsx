import React from 'react';

import MessageInput from 'Components/MessageInput';

jest.mock('@material-ui/core/TextField', () => global.mockComponent('TextField'));

const props = {
  onSendMessage: jest.fn(),
};

describe('render MessageInput component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = global.mount(
      <MessageInput {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('MessageInput').length).toBe(1);
  });

  test('TextField subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'TextField' }).length).toBe(1);
  });
});

describe('snapshot-test MessageInput component', () => {
  test('Renders correct properties', () => {
    global.mountExpect(
      <MessageInput {...props} />,
    ).toMatchSnapshot();
  });
});
