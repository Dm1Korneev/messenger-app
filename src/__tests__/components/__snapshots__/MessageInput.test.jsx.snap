// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test MessageInput component Renders correct properties 1`] = `
<WithStyles(MessageInput)
  onSendMessage={[MockFunction]}
>
  <MessageInput
    classes={
      Object {
        "Input": "MessageInput-Input-2",
        "InputLabel": "MessageInput-InputLabel-1",
        "textField": "MessageInput-textField-3",
      }
    }
    onSendMessage={[MockFunction]}
  >
    <Component
      InputLabelProps={
        Object {
          "classes": Object {
            "root": "MessageInput-InputLabel-1",
          },
        }
      }
      InputProps={
        Object {
          "classes": Object {
            "input": "MessageInput-Input-2",
          },
        }
      }
      className="MessageInput-textField-3"
      color="primary"
      fullWidth={true}
      label="Write a message..."
      margin="normal"
      multiline={true}
      onChange={[Function]}
      onKeyPress={[Function]}
      rows="2"
      rowsMax="5"
      value=""
      variant="outlined"
    >
      <div
        originalcomponent="TextField"
        testprops={
          Object {
            "InputLabelProps": Object {
              "classes": Object {
                "root": "MessageInput-InputLabel-1",
              },
            },
            "InputProps": Object {
              "classes": Object {
                "input": "MessageInput-Input-2",
              },
            },
            "className": "MessageInput-textField-3",
            "color": "primary",
            "fullWidth": true,
            "label": "Write a message...",
            "margin": "normal",
            "multiline": true,
            "onChange": [Function],
            "onKeyPress": [Function],
            "rows": "2",
            "rowsMax": "5",
            "value": "",
            "variant": "outlined",
          }
        }
      />
    </Component>
  </MessageInput>
</WithStyles(MessageInput)>
`;
